kcp/controllers/watcher_controller.go

```diff
@@ -266,15 +238,15 @@ func (r *WatcherReconciler) createOrUpdateIstioVirtualServiceForCR(ctx context.C
 	vsName := obj.GetName()
 	listenerVirtualService, apiErr := istioClientSet.NetworkingV1beta1().
 		VirtualServices(namespace).Get(ctx, vsName, metav1.GetOptions{})
-	ready, err := util.IstioResourcesErrorCheck(istioGatewayGVR, apiErr)
+	ready, err := util.IstioResourcesErrorCheck(util.IstioVirtualServiceGVR, apiErr)
 	if !ready {
 		return err
 	}
 	if errors.IsNotFound(apiErr) {
 		vs := &istioclientapiv1beta1.VirtualService{}
 		vs.SetName(vsName)
 		vs.SetNamespace(namespace)
-		util.UpdateVirtualServiceConfig(vs, obj, istioGatewayResourceName)
+		util.UpdateVirtualServiceConfig(vs, obj, IstioGatewayResourceName, IstioGatewayNamespace)
 		_, err := istioClientSet.NetworkingV1beta1().
 			VirtualServices(namespace).Create(ctx, vs, metav1.CreateOptions{})
 		if err != nil {
```

Comment:
```suggestion
		return err
```

Reply:
this issue is still to be seen at places. Could you please check and remove this?